File: 백준/Silver/15649. N과 M （1）/N과 M （1）.py
import sys
input = sys.stdin.readline

n, m = map(int, input().split())

def main():
    li = []
    def recur():
        if len(li) == m:
            print(*li)
            return
        
        for i in range(1, n + 1):
            if i in li:
                continue
            li.append(i)
            recur()
            li.pop()
    
    recur()
    
    
main()
    

File: 백준/Silver/15649. N과 M （1）/N과 M （1）.js
const fs = require('fs');
const [n, k] = fs.readFileSync("/dev/stdin").toString().trim().split(" ").map(Number);

const li = []
const recur = () => {
    if (li.length == k){
        console.log(li.join(" "));
        return
    }
        
    
    for(let i = 1; i <= n; i++){
        if (li.includes(i)) continue;
        
        li.push(i)
        recur()
        li.pop()
    }    
}

recur();
